lib/Logger.d.ts

```diff
@@ -0,0 +1,65 @@
+declare type Parameters = string | Record<string, unknown>;
+declare type ServerLoggingCallback = (logger: Logger, options: Record<string, unknown>) => Promise<void>;
+declare type ClientLoggingCallBack = (message: string) => void;
+export default class Logger {
+    logLines: Array<Record<string, unknown>>;
```

Comment:
[this line](https://github.com/Expensify/expensify-common/blob/f9985d572c772757885ff59ea5c82b1050c6b3f7/lib/Logger.jsx#L56) gives you more information about what an item in `logLines` looks like.

```
        const length = this.logLines.push({
            message,
            parameters,
            onlyFlushWithOthers,
            timestamp: (new Date())
        });
```

This is the only place where items are added to log lines so you don't have to make properties optional.